File: Domain/Course.cs
namespace Domain
{
    public class Course : BaseEntity
    {
        public required string Name { get; set; }
        public required string Title { get; set; }
        public string? Description { get; set; }
        public string? ImageUrl { get; set; }
        public float Price { get; set; }
        public bool IsPublished { get; set; }

        public Guid? CategoryId { get; set; }
        public Category? Category { get; set; }

        public ICollection<Chapter> Chapters { get; set; } = [];

        public ICollection<Attachment> Attachments { get; set; } = [];

        public ICollection<Purchase> Purchases { get; set; } = [];
    }
}

File: angular-client-app/src/app/shared/types/course.ts
import { Attachment, Chapter } from '.';

export type Course = {
  id: string;
  title: string;
  description?: string;
  imageUrl?: string;
  price?: number;
  isPublished?: boolean;
  categoryId?: string;
  category?: string;
  chapters: Chapter[];
  attachments: Attachment[];
  isPurchased?: boolean;
  userProgress: number;
};
